### DISCUSSION TITLE
Installing turbo on a windows docker container

### DISCUSSION THREAD

**Discussion by rodebeer** [vercel/turborepo#5481] [Help] [answered]
Hi All,

I am tryring to setup Turbo on a windows docker environment for developement prurposes.
I am using 16.15.1-win-x64
Currently I have the following dockerfile:
\# escape=`

\#
\# Development-only image for running Next.js in a containerized environment.
\# Assumes that the Next.js rendering host source is mounted to c:\app.
\#

ARG PARENT_IMAGE
FROM ${PARENT_IMAGE} as debug

USER ContainerAdministrator \#I needed to switch to ConatinerAdministrator, otherwise SymLinks could not be created.
WORKDIR /app

EXPOSE 3000

RUN npm install --location=globa npm@latest
RUN npm install --location=global turbo

\# Add entry point execution file
COPY .\tools\entrypoints\turbo\Development.cmd C:\tools\entrypoints\turbo\Development.cmd

The entry point cmd looks like the following:
echo "Development ENTRYPOINT: starting..."

cd "/app"
echo "Installing npm packages"
npm install
echo "Starting Turbo"
turbo run start:connected

However when building the docker image, I get the following error:
npm ERR! code 1
npm ERR! path C:\node\node_modules\turbo
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c node install.js
npm ERR! node:internal/errors:841
npm ERR!   const err = new Error(message);
npm ERR!               ^
npm ERR!
npm ERR! Error: Command failed: node C:\node\node_modules\turbo\bin\turbo --version --skip-infer
npm ERR!     at checkExecSyncError (node:child_process:828:11)
npm ERR!     at Object.execFileSync (node:child_process:863:15)
npm ERR!     at validateBinaryVersion (C:\node\node_modules\turbo\install.js:28:6)
npm ERR!     at C:\node\node_modules\turbo\install.js:314:7 {
npm ERR!   status: 3221225781,
npm ERR!   signal: null,
npm ERR!   output: [ null, Buffer(0) [Uint8Array] [], Buffer(0) [Uint8Array] [] ],
npm ERR!   pid: 1632,
npm ERR!   stdout: Buffer(0) [Uint8Array] [],
npm ERR!   stderr: Buffer(0) [Uint8Array] []
npm ERR! }

Can anyone explain this error or give some help to get the container up and running?

All help is apriciated.
Regards,
Frank

**Comment by rodebeer** [comment]
I tried to switch to yarn, but this will give me a simular issue:
error C:\Users\ContainerAdministrator\AppData\Local\Yarn\Data\global\node_modules\turbo: Command failed.
Exit code: 1
Command: node install.js
Arguments:
Directory: C:\Users\ContainerAdministrator\AppData\Local\Yarn\Data\global\node_modules\turbo
Output:
node:internal/errors:841
  const err = new Error(message);
              ^

Error: Command failed: node C:\Users\ContainerAdministrator\AppData\Local\Yarn\Data\global\node_modules\turbo\bin\turbo --version --skip-infer
    at checkExecSyncError (node:child_process:828:11)
    at Object.execFileSync (node:child_process:863:15)
    at validateBinaryVersion (C:\Users\ContainerAdministrator\AppData\Local\Yarn\Data\global\node_modules\turbo\install.js:28:6)
    at C:\Users\ContainerAdministrator\AppData\Local\Yarn\Data\global\node_modules\turbo\install.js:314:7 {
  status: 3221225781,
  signal: null,
  output: [ null, Buffer(0) [Uint8Array] [], Buffer(0) [Uint8Array] [] ],
  pid: 1644,
  stdout: Buffer(0) [Uint8Array] [],
  stderr: Buffer(0) [Uint8Array] []

**Comment by anthonyshew** [comment, marked as answer]
If you're still seeing this problem, please open an issue. 🙏
